### DISCUSSION TITLE
Budibase DB table does not load with 4500 records

### DISCUSSION THREAD

**Discussion by daviwall** [Budibase/budibase#2746] [Help] [answered]
Hello everyone,

I opened this topic last week -> https://github.com/Budibase/budibase/discussions/2653 
It is about the issue of importing CSV files during table creation. I was informed that I could use the API to add new records from the CSV. 
I created a python script and was able to add all CSV entries to the table I created. The problem is that now when I click on the table, the system hangs and the data is not shown. Instead I get an error "Error fetching deployment history". The whole system becomes unusable, the design area, the "play" button, everything just hangs even after restarting the entire VM.
Is anyone experiencing this issue with many records? has anyone reported such issue?

The table contains around 4500 records and 40 fields.

Thank you.

**Comment by daviwall** [comment, marked as answer]
Here is an update after some tests.

If you have a single table, when you login as admin/develop, this only table is selected automatically thus freezing the application. So here is what I did to circumvent the issue:

1) Created an empty table starting with letter A (so it stays as first in the list)
2) Created the table I want and imported csv file with 10 records
3) Configured my form (automatically generated with some manual changes afterwards)
4) Import all the data using REST API

These steps prevented me from getting the application hanging every time I log in.
After importing all records, the application is running, BUT it is REALLY slow. It takes 12 seconds to list the records. It takes around the same amount of time to load the form once clicking in "add record".

Any lights on the slowness issue?

Thank you.

**Reply by mjashanks** [reply]
Hey @daviwall - thanks for reporting this, it shouldn't happen.

We will look into this.

**Reply by daviwall** [reply]
Hello @mjashanks thanks for your response.
As I see there are 2 issues:

- When click on a table with 4500 records in the budibase db table area, the system hangs
- A table list or a form takes around 10 seconds to load with 4500 records even though it is paginated with a limit of 10 records

Are you creating issues about these findings? do you need help reproducing them?

**Reply by mjashanks** [reply]
Just to followup on this. We are working on solving this issue at the moment. The issue is happening because we are not paginating the data properly. Then, Budibase is running a request for each row. So - there are 4501 DB requests all firing at once!

We are fixing this at the moment - https://github.com/Budibase/budibase/issues/2632

**Reply by daviwall** [reply]
Thanks for the follow up! Looking forward for the new release :)
